otelcol/collector_test.go

```diff
@@ -476,6 +476,25 @@ func TestPassConfmapToServiceFailure(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestCollectorConfmapLogs(t *testing.T) {
+	set := CollectorSettings{
+		BuildInfo:              component.NewDefaultBuildInfo(),
+		Factories:              nopFactories,
+		ConfigProviderSettings: newDefaultConfigProviderSettings([]string{filepath.Join("testdata", "otelcol-nop.yaml")}),
+	}
+	col, err := NewCollector(set)
+	require.NoError(t, err)
+
+	wg := startCollector(context.Background(), t, col)
+
+	col.Shutdown()
+	wg.Wait()
+	assert.Equal(t, StateClosed, col.GetState())
+
+	assert.NotNil(t, col.ol)
+	assert.True(t, col.ol.Len() > 0)
```

Comment:
It should be fairly easy to create a confmap Provider that always logs something when it's used, then pass it to `CollectorSettings`. We could probably also assert on the log message since we could control it.

Reply:
Could we create a slice of strings containing the `Message` field on each log, then just use `assert.Contains(t, messages, "this is a test log")`? I agree checking a single index feels fragile without also asserting that there is only a single log.

Reply:
Updated